google/cloud/security/notifier/pipelines/base_notification_pipeline.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base pipeline to perform notifications"""
+
+import abc
+
+# TODO: Investigate improving so we can avoid the pylint disable.
+# pylint: disable=line-too-long,no-name-in-module
+from google.cloud.security.common.data_access import dao
+from google.cloud.security.common.data_access import project_dao
+from google.cloud.security.common.data_access import violation_dao
+from google.cloud.security.common.util import log_util
+# pylint: enable=line-too-long,no-name-in-module
+
+LOGGER = log_util.get_logger(__name__)
+
+# pylint: disable=too-many-instance-attributes
+class BaseNotificationPipeline(object):
+    """Base pipeline to perform notifications"""
+
```

Comment:
QQ: Did you wanted to add ``` __metaclass__ = abc.ABCMeta``` here to enforce the abstractmethods are implemented by the specific instance classes?